### PR TITLE
Add missing System.Net docs

### LINKED ISSUE #72403: [System.Net.Http.Json] Undocumented 7.0 APIs in System.Net.Http.Json
@dotnet/ncl 

Below is the list of APIs that were introduced in 7.0 and have not yet been fully documented.

Relevant dates:
-	The last day to merge API documentation directly into triple slash comments in dotnet/runtime is before the RC1 snap on August 15th.
-	After the RC1 snap, all pending documentation changes need to be merged directly into dotnet-api-docs, and the last day to do so is before the RC2 snap on September 13th.

The latest table of undoc APIs, as of Preview6, can be found in [this comment](https://github.com/dotnet/runtime/issues/72403#issuecomment-1211350991).

<details>
  <summary>Old table (as of Preview5, all documented now)</summary>

| Summary  | Params  | TypeParams | Returns | Undocumented API                                                                                                                                                                                                                                       | Docs URL                                                                                                                 |
|----------|---------|------------|---------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------|
| Missing  | Missing | NA         | NA      | [M:System.Net.Http.StringContent.#ctor(System.String,System.Net.Http.Headers.MediaTypeHeaderValue)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.Http/StringContent.xml)                                            | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.Http.StringContent.-ctor?branch=main                       |
| Missing  | Missing | NA         | NA      | [M:System.Net.Http.StringContent.#ctor(System.String,System.Text.Encoding,System.Net.Http.Headers.MediaTypeHeaderValue)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.Http/StringContent.xml)                       | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.Http.StringContent.-ctor?branch=main                       |
| Missing  | Missing | NA         | NA      | [M:System.Net.Http.Headers.MediaTypeHeaderValue.#ctor(System.String,System.String)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.Http.Headers/MediaTypeHeaderValue.xml)                                             | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.Http.Headers.MediaTypeHeaderValue.-ctor?branch=main        |
| Missing  | Missing | NA         | Missing | [M:System.Net.NetworkInformation.Ping.Send(System.String,System.TimeSpan,System.Byte[],System.Net.NetworkInformation.PingOptions)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.NetworkInformation/Ping.xml)        | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.NetworkInformation.Ping.Send?branch=main                   |
| Missing  | Missing | NA         | Missing | [M:System.Net.NetworkInformation.Ping.Send(System.Net.IPAddress,System.TimeSpan,System.Byte[],System.Net.NetworkInformation.PingOptions)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.NetworkInformation/Ping.xml) | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.NetworkInformation.Ping.Send?branch=main                   |
| Missing  | Missing | NA         | NA      | [M:System.Net.Sockets.NetworkStream.Close(System.TimeSpan)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.Sockets/NetworkStream.xml)                                                                                 | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.Sockets.NetworkStream.Close?branch=main                    |
| Missing  | NA      | NA         | NA      | [P:System.Net.Sockets.SafeSocketHandle.IsInvalid](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.Sockets/SafeSocketHandle.xml)                                                                                        | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.Sockets.SafeSocketHandle.IsInvalid?branch=main             |
| Missing  | Missing | NA         | Missing | [M:System.Net.Sockets.Socket.Poll(System.TimeSpan,System.Net.Sockets.SelectMode)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.Sockets/Socket.xml)                                                                  | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.Sockets.Socket.Poll?branch=main                            |
| Missing  | Missing | NA         | NA      | [M:System.Net.Sockets.Socket.Select(System.Collections.IList,System.Collections.IList,System.Collections.IList,System.TimeSpan)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.Sockets/Socket.xml)                   | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.Sockets.Socket.Select?branch=main                          |
| Missing  | NA      | NA         | NA      | [P:System.Net.Sockets.UnixDomainSocketEndPoint.AddressFamily](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.Sockets/UnixDomainSocketEndPoint.xml)                                                                    | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.Sockets.UnixDomainSocketEndPoint.AddressFamily?branch=main |
| Missing  | Missing | NA         | Missing | [M:System.Net.Sockets.UnixDomainSocketEndPoint.Create(System.Net.SocketAddress)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.Sockets/UnixDomainSocketEndPoint.xml)                                                 | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.Sockets.UnixDomainSocketEndPoint.Create?branch=main        |
| Missing  | NA      | NA         | Missing | [M:System.Net.Sockets.UnixDomainSocketEndPoint.Serialize](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.Sockets/UnixDomainSocketEndPoint.xml)                                                                        | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.Sockets.UnixDomainSocketEndPoint.Serialize?branch=main     |
| Missing  | NA      | NA         | Missing | [M:System.Net.Sockets.UnixDomainSocketEndPoint.ToString](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Net.Sockets/UnixDomainSocketEndPoint.xml)                                                                         | https://review.docs.microsoft.com/en-us/dotnet/api/System.Net.Sockets.UnixDomainSocketEndPoint.ToString?branch=main      |

</details>

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/runtime/issues/72403

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet/ncl, can someone review this?

File: src/libraries/System.Net.Http/src/System/Net/Http/Headers/MediaTypeHeaderValue.cs [hunk 1]
```diff
@@ -47,14 +50,18 @@ public class MediaTypeHeaderValue : ICloneable
...
                     else
                     {
-                        Parameters.Add(new NameValueHeaderValue(charSet, value));
+                        Parameters.Add(new NameValueHeaderValue(CharSetName, value));
                     }
                 }
             }
         }
 
+        /// <summary>Gets or sets the media-type header value parameters.</summary>
```
Comment (rzikm) [inline src/libraries/System.Net.Http/src/System/Net/Http/Headers/MediaTypeHeaderValue.cs:59]: The property seems to be get-only
```suggestion
        /// <summary>Gets the media-type header value parameters.</summary>
```

File: src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs [hunk 2]
```diff
@@ -208,6 +358,34 @@ public PingReply Send(string hostNameOrAddress, int timeout, byte[] buffer, Ping
...
+        /// </summary>
+        /// <param name="address">
+        /// An <see cref="IPAddress"/> that identifies the computer that is the destination for the ICMP echo message.
+        /// </param>
+        /// <param name="timeout">
+        /// An <see cref="int"/> value that specifies the maximum number of milliseconds (after sending the echo message) to wait for the ICMP echo reply message.
+        /// </param>
+        /// <param name="buffer">
+        /// A <see cref="byte"/> array that contains data to be sent with the ICMP echo message and returned in the ICMP echo reply message.
+        /// The array cannot contain more than 65,500 bytes.
```
Comment (rzikm) [inline src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs:373]: here you use 65,500, but the ArgumentException doc comment uses 65500 so there is some inconsistency (same for other overloads).

**Review by rzikm — APPROVED** [review]
LGTM apart from minor nits.

File: src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs (hunk 2 above)
Comment (stephentoub) [reply src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs:373]: Fixed, thanks

**Pull request merged**